Amplify/Categories/DataStore/Model/DataType/DateScalarOperation.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+extension DateScalar {
+
+    ///
+    /// - Parameters:
+    ///   - value: the amount to add, or subtract in case of negative values
+    ///   - component: the component that will get the value added
+    /// - Returns: a new instance of the current DateScalar type
+    func add(value: Int, to component: Calendar.Component) -> Self {
+        let calendar = Self.iso8601Calendar
+        let result = calendar.date(byAdding: component, value: value, to: self.date)
+        guard let date = result else {
+            fatalError(
```

Comment:
I updated the error.

I had a really hard time understanding why that API returns an optional. It is very hard to reproduce that scenario given our implementation is guarded by the ISO8601 format which guarantees the necessary fields will be there.

My take on this right now is to make the message clear and observe if developers will run into this to understand the scenarios where it will occur better.

Making the API throw an error or return optionals will make it not so convenient to use, which is the primary goal (as opposed to the default calendar operations).